src/indicators.ts

```diff
@@ -59,19 +60,26 @@ export class Indicators {
         return new Promise<Section>((resolve, reject) => {
             this.xmlParse.parseContent(xmlFile, (err, data) => {
                 if (err) { return reject(err); }
-                const section = data.find((lcovSection) => {
-                    // consider windows and linux file paths
-                    let cleanFile = file.replace(/[\\\/]/g, "");
-                    let cleanLcovFileSection = lcovSection.file.replace(/[\\\/]/g, "");
-
-                    // on Windows remove drive letter from path because of cobertura format
-                    // also convert both path to lowercase because Windows's filesystem is case insensitive
-                    if (process.platform === "win32") {
-                        cleanFile = cleanFile.substr(2).toLowerCase();
-                        cleanLcovFileSection = cleanLcovFileSection.toLowerCase();
-                    }
 
-                    return cleanFile.includes(cleanLcovFileSection);
+                const cleanFileToMatch = file.split(/[\\\/]/).reverse();
+
+                // Find best path match
+                let bestMatchLength = 0;
+                let section = null;
+
+                _.forEach(data, (xmlSection) => {
+                    const cleanLcovFileSection = xmlSection.file.split(/[\\\/]/).reverse();
+
+                    // Comparing length of shared path
+                    const zippedPaths = _.zip(cleanFileToMatch, cleanLcovFileSection);
+                    let sharedPath = _.map(zippedPaths, (x) => x[0] === x[1]);
+                    sharedPath = _.compact(sharedPath);
+
+                    // New best match
+                    if (sharedPath.length > bestMatchLength) {
+                        section = xmlSection;
+                        bestMatchLength = sharedPath.length;
+                    }
```

Comment:
this seems very similar to my leaf node compare function (for the lcov version) in this same file https://github.com/ryanluker/vscode-coverage-gutters/pull/120/files#diff-fc82abb2ab9c0fe32935ce85f70da9a7R147 . I am currently working on a refactor of this file for the 2.0.0 release so I will probably standardize the xml and lcov code flows based on your implementation 😃 .

Reply:
I found it hard to understand.

Besides it was used in `lcovExtract` and not in `xmlExtract`. Expanding it with match counter could break `lcovExtract` and I don't even have any lcov coverage to test it properly ;)

Reply:
yep like I mentioned I am in here anyways with my 2.0 refactor so i can handle the conversion.